libcodechecker/libclient/product_helper.py

```diff
@@ -0,0 +1,106 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+
+import os
+import sys
+# import datetime
+import socket
+
+from thrift.transport import THttpClient
+from thrift.protocol import TJSONProtocol
+from thrift.protocol.TProtocol import TProtocolException
+
+import shared
+from ProductManagement import productService
+
+from libcodechecker import session_manager
+
+
+class ThriftProductHelper(object):
+    def __init__(self, host, port, uri, session_token=None):
+        self.__host = host
+        self.__port = port
+        self.transport = THttpClient.THttpClient(self.__host, self.__port, uri)
+        self.protocol = TJSONProtocol.TJSONProtocol(self.transport)
+        self.client = productService.Client(self.protocol)
+
+        if session_token:
+            headers = {'Cookie': session_manager.SESSION_COOKIE_NAME +
+                       "=" + session_token}
+            self.transport.setCustomHeaders(headers)
+
+            # ------------------------------------------------------------
```

Comment:
The reason is unknown. I think back when doing the _Authentication_ module we decided to keep the pattern. Also, different endpoint clients _could_ have different ways of handling the received exceptions. I think there are subtle differences in the `except` statements already across the modules.

I would rather not introduce extra refactorings into this package. The scheme was already given and existing in the code, and I like keeping that.